src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/Models/ReindexJobRecord.cs

```diff
@@ -52,7 +52,7 @@ public class ReindexJobRecord : JobRecord
             // check for MaximumConcurrency boundary
             if (maxiumumConcurrency > MaxMaximumConcurrency)
             {
-                throw new BadRequestException(string.Format(Fhir.Core.Resources.InvalidReIndexParameterValue, nameof(MaximumConcurrency), MinMaximumConcurrency.ToString(), MaxMaximumConcurrency.ToString()));
+                throw new ArgumentOutOfRangeFhirException(string.Format(Fhir.Core.Resources.InvalidReIndexParameterValue, nameof(MaximumConcurrency), MinMaximumConcurrency.ToString(), MaxMaximumConcurrency.ToString()));
```

Comment:
>ArgumentOutOfRangeFhirException [](http://example.com/codeflow?start=26&length=31)

Can I ask what value it adds?

From what I see we already can achive what we want with combination of existing exception and text message inside it.

What's the point to add new exception?

Reply:
I see what your point is and I agree but the idea behind this is to club all the Argument value constraints together for the better distinction

Reply:
>to club all the Argument value constraints 

Aren't they already club together inside this class?

Adding something new usually justified by fact what it allows to achive something new, for the price of increasing entropy of the code.

From changes in this PR, I don't think there is anything new in behavior, only difference would be type of exception been thrown, and I'm not sure we care is it BadRequest or ArgumentOut exception.

Reply:
hmm I see your concern. if we are fine to stick with BadException type then I am good with it ☺